pkg/reconciler/pipelinerun/resources/apply_test.go

```diff
@@ -3599,9 +3601,34 @@ func TestApplyTaskResultsToPipelineResults(t *testing.T) {
 			Name:  "pipeline-result-2",
 			Value: *v1beta1.NewStructuredValues("do, rae, mi, rae, do"),
 		}},
+	}, {
+		description: "multiple-results-skipped-and-normal-tasks",
+		results: []v1beta1.PipelineResult{{
+			Name:  "pipeline-result-1",
+			Value: *v1beta1.NewStructuredValues("$(tasks.skippedTask.results.foo)"),
+		}, {
+			Name:  "pipeline-result-2",
+			Value: *v1beta1.NewStructuredValues("$(tasks.skippedTask.results.foo), $(tasks.normaltask.results.baz)"),
+		}},
+		taskResults: map[string][]v1beta1.TaskRunResult{
+			"normaltask": {{
+				Name:  "baz",
+				Value: *v1beta1.NewStructuredValues("rae"),
+			}},
+		},
+		skippedTasks: []v1beta1.SkippedTask{{
+			Name: "skippedTask",
+		}},
+		expectedResults: []v1beta1.PipelineRunResult{{
```

Comment:
What output results existed prior to this regression? Was it just a non-replaced result expression? or was there no pipeline result? (based on the linked issue it looks like the latter.) I think an absent Pipeline result might make more sense than a Pipeline result like "$(tasks.skippedTask.results.foo)".

(everything else is mostly a nit but let's verify this)

Reply:
Ah yes this is a great point! When I implemented this fix I didn't check the previous behaviour. I checked the previous version and there are no results if the task is skipped. We should definitely keep the previous behaviour otherwise this will be another issue